src/lib/input/input-container.ts

```diff
@@ -250,12 +264,7 @@ export class MdInputDirective {
   /** Whether the input is in an error state. */
   _isErrorState(): boolean {
     const control = this._ngControl;
-    const isInvalid = control && control.invalid;
-    const isTouched = control && control.touched;
-    const isSubmitted = (this._parentFormGroup && this._parentFormGroup.submitted) ||
-        (this._parentForm && this._parentForm.submitted);
-
-    return !!(isInvalid && (isTouched || isSubmitted));
+    return this.errorStateMatcher(control, this._parentFormGroup, this._parentForm);
```

Comment:
I think we should be passing through `this._ngControl.control` here. In other words, the `FormControl` instance rather than the `NgControl` directive. I don't think most people are familiar with `NgControls`, and probably aren't familiar with the subtle distinction between the directive and the `FormControl` itself. While most properties are mirrored on the directive, a few aren't, and I foresee that being confusing.